src/util.h

```diff
@@ -662,5 +662,17 @@ inline std::vector<std::wstring_view> split(const wchar_t* s, const wchar_t* del
     return split(std::wstring_view(s), std::wstring_view(delim));
 }
 
+/**
+ * Helper class that runs a function at destruction.
+ */
+class Deferred {
+public:
+    Deferred(std::function<void()> deferred) : closer(std::move(deferred)) {}
+    ~Deferred() { closer(); }
+
+private:
+    std::function<void()> closer;
```

Comment:
Ah, nit, sorry, "closer" I think comes from the "socket closer function" in ZeroMQ, maybe call that member `deferred`, too?

Reply:
Yeah, I just copied and pasted it as a whole 😛